macros/src/path.rs

```diff
@@ -0,0 +1,244 @@
+
+    #[ts(export_to = env(\"MY_ENV_VAR_PATH\"))] 
+
+Note: This option is for environment variables defined in the '.cargo/config.toml' file only, accessible through the `env!` macro!
+";
+        let get_path = |input: ParseStream| -> Result<(syn::Path,Option<LitStr>)>{ 
+            let mut tokens = TokenStream::new();
+            let mut env_var_str = None;
+
+            if input.peek(Token![self])  {
+                let token = input.parse::<Token![self]>()?;
+                tokens.extend(quote!(#token));
+            }
+            if input.peek(Token![super]) {
+                let token = input.parse::<Token![super]>()?;
+                tokens.extend(quote!(#token));
+            }
+            if input.peek(Token![crate]) {
+                let token = input.parse::<Token![crate]>()?;
+                tokens.extend(quote!(#token));
+            }
+            if input.peek(Ident) {
+                let ident = input.parse::<Ident>()?;
+                tokens.extend(quote!(#ident));
+            }
+
+            while input.peek(Token![::]) {
+                let token = input.parse::<Token![::]>()?;
+                tokens.extend(quote!(#token));
+
+                if input.peek(Ident){
+                    let ident = input.parse::<Ident>()?;
+                    tokens.extend(quote!(#ident));
+                } else { return Err(Error::new(input.span(),"expected ident")) }
+            }
+
+            if input.peek(syn::token::Paren){
+                let content;
+                syn::parenthesized!(content in input);
+                env_var_str = Some(content.parse::<LitStr>()?);
+            }
+
+            Ok((syn::parse2::<syn::Path>(tokens)?,env_var_str))
+        };
+
+
+        // string literal
+        if input.peek(LitStr){
+            if let Ok(lit) = Lit::parse(input){
+                match lit {
+                    Lit::Str(string) => { return Ok(CustomPath::Str(string.value())); },
+                    _ => { return Err(Error::new(span, msg)); },
+                }
+            } 
+        } 
+
+        match get_path(input) {
+
+            Ok((path,arg)) => {
+    
+                if !path.segments.is_empty(){
+        
+                    if let Some( env_var_str ) = arg {
+            
+                        if path.is_ident("env") {
+                            return Ok(CustomPath::Env(env_var_str));
+                        }
+         
+                    } else {
+        
+                        let last  = &path.segments.last().unwrap().ident;
+        
+                        // static or const 
+                        if is_screaming_snake_case(&last.to_string()) {
+                            return Ok(CustomPath::Static(path));
+                        }
+            
+                        // function 
+                        if is_snake_case(&last.to_string()) {
+                            return Ok(CustomPath::Fn(path));
+                        }    
```

Comment:
Another reason I don't really like allowing a function path here is that there is no reliable way to differentiate between a function name and a const/static name. Sure, clippy will complain if a const isn't in SCREAMING_SNAKE_CASE, but it doesn't prevent the code from compiling

Reply:
I've triyed  this scenario on the new design (I've mentioned above, assignment to a function pointer ) and it looks like 
it will not compile, which is not bad, this means that the macro is limited to 
Rust naming conventions. As long as the mistake is excluded we are ok.

so for lower case static the error will be : 

```terminal 
expected fn pointer `fn(&'static str) -> std::option::Option<&'static str>`
    found reference `&'static str`
```